monitoring/nethealth.go

```diff
@@ -262,38 +263,59 @@ func (c *nethealthChecker) isHealthy(peer string) (healthy bool, err error) {
 // fetchNethealthMetrics collects the network metrics from the nethealth pod
 // specified by addr. Returns mapping of peer to networkData.
 func fetchNethealthMetrics(ctx context.Context, addr string) (map[string]networkData, error) {
-	client, err := roundtrip.NewClient(addr, "")
-	if err != nil {
-		return nil, trace.Wrap(err, "failed to connect to nethealth service at %s.", addr)
-	}
-
-	// The two relevant metrics exposed by nethealth are 'nethealth_echo_request_total' and
-	// 'nethealth_echo_timeout_total'. We expect a pair of request/timeout metrics per peer.
-	// Example metrics received from nethealth may look something like the output below:
-	//
-	//      # HELP nethealth_echo_request_total The number of echo requests that have been sent
-	//      # TYPE nethealth_echo_request_total counter
-	//      nethealth_echo_request_total{node_name="10.128.0.96",peer_name="10.128.0.70"} 236
-	//      nethealth_echo_request_total{node_name="10.128.0.96",peer_name="10.128.0.97"} 273
-	//      # HELP nethealth_echo_timeout_total The number of echo requests that have timed out
-	//      # TYPE nethealth_echo_timeout_total counter
-	//      nethealth_echo_timeout_total{node_name="10.128.0.96",peer_name="10.128.0.70"} 37
-	//      nethealth_echo_timeout_total{node_name="10.128.0.96",peer_name="10.128.0.97"} 0
-	resp, err := client.Get(ctx, client.Endpoint("metrics"), url.Values{})
+	respCh := make(chan *http.Response, 1)
+	errCh := make(chan error, 1)
+	go func() {
```

Comment:
I'm seeing this goroutine as an attempt to handle context expiration and HTTP query simultaneously - is this the case?
Why not use a request with a context for that case (as it sort of was before)? Also, using the default HTTP client is not ideal in most of the cases due to its implicit configuration w.r.t timeouts.

Reply:
Yeah, I wanted to handle an http request with context here without using the `roundtrip` package. 

Gravity 5.5 is using an older version of `roundtrip`. So if we wanted to use `roundtrip` here we would need to update gravity's version of the package and all of the code that imports the package.

I think gravity 6.x has updated the `roundtrip` package, so we can use the previous implementation in versions >= 6.x

Reply:
You don't need roundtrip to make an HTTP request bound to a context:
```go
// create a custom HTTP client
req := http.NewRequestWithContext(ctx)
resp, err := client.Do(req)
```